### PR TITLE
remove hard code of filtering .* hidden file

### PR DESCRIPTION
**Pull request by zyf0330**
remove hard code of filtering .* hidden file
fix improper test_get_os_stat test

### CODE REVIEW

**Comment by bashimao** [comment]
Just wanted to point out that this fix - as simple as it might look - works fine on LinuxMint 18.2. I dropped some git-versioned folders into the onedrive directory. Downloading the files with the Windows 10 client from Microsoft on the other machine worked flawlessly as well. Restarting the onedrive-d daemon did also not yield any undesired effects.

**Comment by zyf0330** [comment]
😀

**Comment by bogdanm** [comment]
+1 to merge this. It's creating loads of issues in Linux for me (none of which are present in Windows).

**Comment by derrix060** [comment]
I would recommend you to create tests for this change. See my comments on the commit.

File: tests/test_tasks.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def test_rename_with_suffix_and_count(self):
         self.assertTrue(os.path.isfile(self.repo.local_root + '/foo 3 (hostname).txt'))
 
     def test_get_os_stat(self):
-        self.assertIsNone(merge_dir.get_os_stat('/foo/bar/baz/blah'))
+        self.assertIsNone(merge_dir.get_os_stat('/tmp/random_file_sdhfuhsaiu123blah'))
```
[the remark this answers is not in the document]
Comment (zyf0330) [reply tests/test_tasks.py]: Test fails, give this
```
ERROR: test_get_os_stat (tests.test_tasks.TestMergeDirTask)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zyf/app/onedrived-dev/tests/test_tasks.py", line 114, in test_get_os_stat
    self.assertIsNone(merge_dir.get_os_stat('/foo/bar/baz/blah'))
  File "/home/zyf/app/onedrived-dev/onedrived/od_tasks/merge_dir.py", line 45, in get_os_stat
    return os.stat(path)
NotADirectoryError: [Errno 20] Not a directory: '/foo/bar/baz/blah'
```

**Comment by zyf0330** [comment]
@derrix060 I have added test, how do you think??

File: tests/test_tasks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xybu) [reply tests/test_tasks.py]: if it's broken before the PR, it should be fixed separately.

[the remark this answers is not in the document]
Comment (zyf0330) [reply tests/test_tasks.py]: Now let us merge this PR first, then fix that bug.

**Comment by kennethreitz** [comment]
+1

**Pull request closed without merging**